lib/bolt/cli.rb

```diff
@@ -313,38 +363,16 @@ def parse
       validate(options)
 
       # After validation, initialize inventory and targets. Errors here are better to catch early.
-      options[:targets] = inventory.get_targets(options[:nodes]) if options[:nodes]
+      unless options[:action] == 'show' || options[:mode] == 'plan'
```

Comment:
This is starting to get ugly, but I'm ok leaving refactoring it for later.

Reply:
I was tempted to move it to the `execute` function, but held back because of the comment.